rbac/rbac/celery.py

```diff
@@ -73,4 +79,14 @@ def start_metrics_server(sender=None, **kwargs):
     """Start the metrics server."""
     registry = CollectorRegistry()
     multiprocess.MultiProcessCollector(registry)
-    start_http_server(LoadedConfig.metricsPort, addr="0.0.0.0", registry=registry)
+    metrics_port = LoadedConfig.metricsPort or 9000
+
+    try:
+        start_http_server(metrics_port, addr="0.0.0.0", registry=registry)
+        logger.info(f"Server started and exposing to port {metrics_port}.")
+        raise Exception("test exception")
+    except Exception as e:
+        sentry_sdk.capture_exception(e)
+        logger.error(f"Failed to start metrics server: {e}")
+        # Exit the entire process, we don't wat to spin up the worker without metrics
+        sys.exit()
```

Comment:
Maybe non-zero code to indicate error?

```suggestion
        sys.exit(1)
```

Reply:
Updated